docs/grammar/Solidity.g4

```diff
@@ -0,0 +1,248 @@
+booleanLiteral: True | False;
+stringLiteral: StringLiteralFragment+;
+hexLiteral: HexLiteralFragment+;
+unicodeStringLiteral: UnicodeStringLiteralFragment+;
+
+numberLiteral: (DecimalNumber | HexNumber) NumberUnit?;
+
+statement:
+    block
+	| simpleStatement
+	| ifStatement
+	| forStatement
+	| whileStatement
+	| doWhileStatement
+	| continueStatement
+	| breakStatement
+	| tryStatement
+	| returnStatement
+	| emitStatement
+    | assemblyStatement
+;
+
+block: LBrace statement* RBrace;
+simpleStatement: variableDeclarationStatement | expressionStatement;
+ifStatement: If LParen expression RParen statement (Else statement)?;
+forStatement: For LParen (simpleStatement | Semicolon) (expressionStatement | Semicolon) expression? RParen statement;
+whileStatement: While LParen expression RParen statement;
+doWhileStatement: Do statement While LParen expression RParen Semicolon;
+continueStatement: Continue Semicolon;
+breakStatement: Break Semicolon;
+tryStatement: Try expression (Returns LParen returnParameters=parameterList RParen)? block catchClause+;
+returnStatement: Return expression? Semicolon;
+emitStatement: Emit expression functionCallArgumentList Semicolon;
+assemblyStatement: Assembly AssemblyDialect? AssemblyLBrace yulStatement* YulRBrace;
+
+variableDeclarationListMaybeEmpty: (Comma* variableDeclarations+=variableDeclaration) (Comma (variableDeclarations+=variableDeclaration)?)*;
+variableDeclarationStatement:
+    (
+        variableDeclaration
+        | (
+            LParen
+                variableDeclarationListMaybeEmpty
+            RParen
+        )
+    ) (Assign expression)? Semicolon;
+expressionStatement: expression Semicolon;
+
+catchClause: Catch (identifier? LParen (arguments=parameterList)? RParen)? block;
+
+mappingType: Mapping LParen key=mappingKey Arrow value=typeName RParen;
+mappingKey: elementaryTypeName | userDefinedTypeName;
+
+yulStatement:
+    yulBlock
+	| yulVarDecl
+	| yulAssignment
+	| yulFunctionCall
+	| yulIfStatement
+	| yulForStatement
+	| yulSwitchStatement
+	| YulLeave
+	| YulBreak
+	| YulContinue
+	| yulFunctionDefinition;
+
+yulBlock: YulLBrace yulStatement* YulRBrace;
+
+yulVarDecl: YulLet variables+=Yulidentifier (YulComma variables+=Yulidentifier)* (YulAssign value=yulExpression)?;
+
+yulAssignment: yulPath YulAssign yulExpression | (yulPath (YulComma yulPath)+) YulAssign yulFunctionCall;
+
+yulIfStatement: YulIf cond=yulExpression body=yulBlock;
+
+yulForStatement: YulFor init=yulBlock cond=yulExpression post=yulBlock body=yulBlock;
+
+yulSwitchCase: YulDefault yulBlock | YulCase yulLiteral yulBlock;
+yulSwitchStatement: YulSwitch yulExpression | yulSwitchCase+;
+
+yulFunctionDefinition:
+	YulFunction Yulidentifier
+	YulLParen (arguments+=Yulidentifier (YulComma arguments+=Yulidentifier)*)? YulRParen
+	((YulArrow | YulMinus YulGreater) returnParameters+=Yulidentifier (YulComma returnParameters+=Yulidentifier)*)?
```

Comment:
Can you add a todo hear to deal with YulArrow/*? Or even just link to the issue.

Reply:
If we decide to fix the issue as bugfix, I could just only use "YulArrow" here :-)...